### DISCUSSION TITLE
Help Needed: How to Redirect a visitor with intended URL if navigation is enabled

### DISCUSSION THREAD

**Discussion by tahirafridi** [livewire/livewire#6354] [1. Help] [answered]
I need to know if this feature is supported or not. In v2 I've used it like
`return redirect()->intended(route('dashboard'));`
In v3 I'm trying to use it like this 
`return $this->redirect(intended(route('dashboard')), navigate: true);`
It says function 'intended' undefined.
The whole story is that I'm trying to achieve, if someone click a link and that link is auth protected, it will redirect to login page and from there if it logged in they will redirect to the intended url not the default one with SPA kind of feeling.

**Comment by sinnbeck** [comment, marked as answer]
Maybe try making a helper function ? This is how the built in function works 
```
function intended($default = '/')
{
   return session()->pull('url.intended', $default);
}
```
Or
```
return $this->redirect(session()->pull('url.intended', route('dashboard')), navigate: true);

**Reply by tahirafridi** [reply]
Thank you @sinnbeck, it resolved my problem.

**Comment by tahirafridi** [comment]
> Maybe try making a helper function ? This is how the built in function works
> 
> ```
> function intended($default = '/')
> {
>    return session()->pull('url.intended', $default);
> }
> ```
> 
> Or
> 
> ```
> return $this->redirect(session()->pull('url.intended', route('dashboard')), navigate: true);
> ```

Thanks @sinnbeck for your code snippet, I will try it and will let you know if it works in my case. Thanks for your time.

**Comment by Crinchy** [comment]
Very cool, it works for me 👍

**Comment by hasan-aftab** [comment]
session()->put('url.intended', request()->getRequestUri());

Or

app('redirect')->setIntendedUrl(request()->getRequestUri());
